python/ray/data/_internal/datasource/lance_datasource.py

```diff
@@ -45,6 +45,7 @@ def __init__(
             self.scanner_options["filter"] = filter
         self.storage_options = storage_options
         self.lance_ds = lance.dataset(uri=uri, storage_options=storage_options)
+        self.projected_schema = self.lance_ds.scanner(**self.scanner_options).projected_schema
```

Comment:
it'd be great if someone from lance could also take a second look (cc @westonpace ?)